### PR TITLE
chore: move storybook-react to apps/ workspace

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This pull request moves the `storybook-react` folder and its files to the `apps/` workspace to better align with our project structure. Additionally, I have updated the `package.json` and `yarn.config` to remove the `docs` workspace, as it is no longer needed.

The changes ensure that the workspace organization is cleaner and avoids maintaining unnecessary references to the `docs` folder.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-design-system/issues/55

## **Manual testing steps**

1. Check that `storybook-react` has been moved to the `apps/` directory
2. Ensure all instances of `docs/storybook-react` have been updated to `apps/storybook-react`
3. Verify that storybook works from root using `yarn workspace @metamask/storybook-react storybook` and `yarn storybook` from root

### **After**

Recording shows storybook still working with both `yarn workspace @metamask/storybook-react storybook` and `yarn storybook` from root

https://github.com/user-attachments/assets/0b4b61a5-3aae-4534-b493-64423a20b6ce

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I’ve completed the PR template to the best of my ability.
- [x] I’ve included tests if applicable.
- [x] I’ve applied the correct labels for this PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
  
## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g., pulled and built the branch, tested the updated folder structure).
- [ ] I confirm that this PR addresses all acceptance criteria described in the linked issue.
- [ ] The app works without any broken references, errors, or regressions after the changes.

### CODE REVIEW

File: apps/storybook-react/vite.config.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
+// import path from 'path';
 import { defineConfig } from 'vite';
-import path from 'path';
```
Comment (georgewrmarshall) [inline apps/storybook-react/vite.config.ts:2]: Use is commented out so commenting it out fixes prettier issue

File: package.json [hunk 2]
```diff
@@ -10,7 +10,6 @@
   "files": [],
   "workspaces": [
     "apps/*",
-    "docs/*",
```
Comment (georgewrmarshall) [inline package.json:13]: Removing docs as a workspace

File: tsconfig.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
 {
   "compilerOptions": { "esModuleInterop": true, "noEmit": true },
   "references": [
-    { "path": "./docs/storybook-react" },
     { "path": "./packages/design-system-react-native" },
     { "path": "./packages/design-system-tailwind-preset" },
     { "path": "./packages/metamask-module-template" },
-    { "path": "./apps/storybook-react-native" }
+    { "path": "./apps/storybook-react-native" },
+    { "path": "./apps/storybook-react" }
```
Comment (georgewrmarshall) [inline tsconfig.json:8]: Reordering so that storybook are together

File: yarn.config.cjs [hunk 4]
```diff
@@ -38,14 +38,13 @@ module.exports = defineConfig({
       const isChildWorkspace = workspace.cwd !== '.';
       const isPrivate =
         'private' in workspace.manifest && workspace.manifest.private === true;
-      const isDocs = workspace.cwd.startsWith('docs/');
```
Comment (georgewrmarshall) [inline yarn.config.cjs:41]: No longer needed as we are removing as a workspace

File: yarn.config.cjs [hunk 5]
```diff
@@ -38,14 +38,13 @@ module.exports = defineConfig({
...
       const isApps = workspace.cwd.startsWith('apps/');
       const dependenciesByIdentAndType = getDependenciesByIdentAndType(
         Yarn.dependencies({ workspace }),
       );
 
-      // Completely ignore docs and apps workspaces
-      if (isDocs || isApps) {
+      // Completely ignore apps workspaces
+      if (isApps) {
```
Comment (georgewrmarshall) [inline yarn.config.cjs:47]: Still ignoring yarn constraints in apps folder

**Review by garrettbear — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -31,6 +30,7 @@
     "prepare-preview-builds": "./scripts/prepare-preview-builds.sh",
     "publish-previews": "yarn workspaces foreach --all --no-private --parallel --verbose run publish:preview",
     "setup": "yarn install",
+    "storybook": "yarn workspace @metamask/storybook-react storybook",
```
Comment (georgewrmarshall) [inline package.json:33]: Adding storybook script to root package.json for better DX

**Pull request merged**